File: src/test/java/workshop/part1/AkkaTest.java
package workshop.part1;

import java.util.concurrent.TimeUnit;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.testkit.TestKit;
import akka.testkit.TestProbe;
import org.junit.After;
import scala.concurrent.duration.Duration;
import scala.concurrent.duration.FiniteDuration;

public abstract class AkkaTest {

    protected ActorSystem system = ActorSystem.create();
    protected TestProbe sender = TestProbe.apply(system);

    @After
    public void teardown() {
        boolean verifySystemShutdown = true;
        TestKit.shutdownActorSystem(system, Duration.create(5, TimeUnit.SECONDS), verifySystemShutdown);
    }

    protected void schedule(FiniteDuration delay, ActorRef receiver, Object msg) {
        system.scheduler().scheduleOnce(delay, receiver, msg, system.dispatcher(), sender.ref());
    }
}

File: src/test/kotlin/workshop/part1/AkkaTest.kt
package workshop.part1

import akka.actor.ActorRef
import akka.actor.ActorSystem
import akka.testkit.TestKit
import akka.testkit.TestProbe
import org.junit.After
import scala.concurrent.duration.Duration
import scala.concurrent.duration.FiniteDuration
import java.util.concurrent.TimeUnit

abstract class AkkaTest {

    protected var system = ActorSystem.create()
    protected var sender = TestProbe.apply(system)

    @After
    fun teardown() {
        val verifySystemShutdown = true
        TestKit.shutdownActorSystem(system, Duration.create(5, TimeUnit.SECONDS), verifySystemShutdown)
    }

    protected fun schedule(delay: FiniteDuration, receiver: ActorRef, msg: Any) {
        system.scheduler().scheduleOnce(delay, receiver, msg, system.dispatcher(), sender.ref())
    }
}
